### PR TITLE
debezium/dbz#815 Add StarRocks JDBC sink support

### LINKED ISSUE #815: Introduce dialect support for Starrocks database [DBZ-6428]
Migrated from [DBZ-6428](https://issues.redhat.com/browse/DBZ-6428)

See https://github.com/StarRocks/starrocks
{noformat}
org.apache.kafka.connect.errors.ConnectException: Exiting WorkerSinkTask due to unrecoverable exception.
at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:614)
at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:329)
at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:232)
at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:201)
at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:185)
at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:234)
at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: org.apache.kafka.connect.errors.ConnectException: JDBC sink connector failure
at io.debezium.connector.jdbc.JdbcSinkConnectorTask.put(JdbcSinkConnectorTask.java:78)
at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:586)
... 10 more
Caused by: org.apache.kafka.connect.errors.ConnectException: Failed to process a sink record
at io.debezium.connector.jdbc.JdbcChangeEventSink.execute(JdbcChangeEventSink.java:71)
at io.debezium.connector.jdbc.JdbcSinkConnectorTask.put(JdbcSinkConnectorTask.java:87)
... 11 more
Caused by: jakarta.persistence.PersistenceException: Converting {{org.hibernate.exception.SQLGrammarException}} to JPA {{PersistenceException}} : JDBC exception executing SQL [CREATE TABLE ods_dw_prod_product_rt (id integer NOT NULL, name longtext NULL, price integer NULL, color longtext NULL, PRIMARY KEY(id))]
at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:165)
at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:175)
at org.hibernate.query.spi.AbstractQuery.executeUpdate(AbstractQuery.java:654)
at io.debezium.connector.jdbc.JdbcChangeEventSink.createTable(JdbcChangeEventSink.java:146)
at io.debezium.connector.jdbc.JdbcChangeEventSink.checkAndApplyTableChangesIfNeeded(JdbcChangeEventSink.java:99)
at io.debezium.connector.jdbc.JdbcChangeEventSink.execute(JdbcChangeEventSink.java:67)
... 12 more
Caused by: org.hibernate.exception.SQLGrammarException: JDBC exception executing SQL [CREATE TABLE ods_dw_prod_product_rt (id integer NOT NULL, name longtext NULL, price integer NULL, color longtext NULL, PRIMARY KEY(id))]
at org.hibernate.exception.internal.SQLExceptionTypeDelegate.convert(SQLExceptionTypeDelegate.java:64)
at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:56)
at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:109)
at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:95)
at org.hibernate.sql.exec.internal.StandardJdbcMutationExecutor.execute(StandardJdbcMutationExecutor.java:97)
at org.hibernate.query.sql.internal.NativeNonSelectQueryPlanImpl.executeUpdate(NativeNonSelectQueryPlanImpl.java:78)
at org.hibernate.query.sql.internal.NativeQueryImpl.doExecuteUpdate(NativeQueryImpl.java:820)
at org.hibernate.query.spi.AbstractQuery.executeUpdate(AbstractQuery.java:643)
... 15 more
Caused by: java.sql.SQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'longtext' at line 1
at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:120)
at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:916)
at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1061)
at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1009)
at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1320)
at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:994)
at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeUpdate(NewProxyPreparedStatement.java:1502)
at org.hibernate.sql.exec.internal.StandardJdbcMutationExecutor.execute(StandardJdbcMutationExecutor.java:84)
... 18 more
{noformat}

The generic ANSI92 SQL dialect that gets chosen by default appears not to be compatible with Starrocks.
It would be great to introduce a dialect for this sink database platform.

Data type reference for Starrocks: https://docs.starrocks.io/en-us/2.5/sql-reference/sql-statements/data-types/VARCHAR

### LINKED ISSUE #979: Support setting target schema for JDBC sink with Starrocks target [DBZ-7824]
Migrated from [DBZ-7824](https://issues.redhat.com/browse/DBZ-7824)

When using the MySQL driver to connect to Starrocks, is puts the user into a schema that is not the desired catalog and there is no way to change this when using the MySQL client. 

To support this a new option {{dialect.starrocks.catalog_name}} is being introduced to allow the user to set the catalog name, and this will be applied before executing the session work.

### PR DESCRIPTION
**Pull request by minleejae**
### Related issues

- Fixes https://github.com/debezium/dbz/issues/815 — Introduce dialect support for StarRocks database
- Fixes https://github.com/debezium/dbz/issues/979 — Support setting target schema for JDBC sink with StarRocks target

### Summary

Adds initial StarRocks support to the Debezium JDBC sink connector, following the structure of the SingleStore support in #7566.

- **Dialect & automatic detection**: StarRocks speaks the MySQL wire protocol but is not DDL-compatible with MySQL, so the Debezium dialect extends `GeneralDatabaseDialect` while a minimal Hibernate `StarRocksDialect` (mirroring Hibernate's `TiDBDialect` precedent) pins the diverging column types. The bundled `com.starrocks:starrocks-connector-j` driver reports the `StarRocks` product name, which a registered Hibernate `DialectResolver` resolves automatically — no `hibernate.dialect` configuration is needed.
- **Upserts & table creation**: StarRocks executes `INSERT` statements against PRIMARY KEY tables as upserts (it has no `ON DUPLICATE KEY UPDATE`), so `insert.mode=upsert` generates plain inserts. Auto-created tables emit `PRIMARY KEY (...) DISTRIBUTED BY HASH (...)` clauses from the record key fields, with `NULL/NOT NULL` placed before `DEFAULT` as StarRocks requires.
- **Type mappings** (verified against StarRocks 4.1.1): time logical types are stored as `HH:mm:ss.ffffff` strings (StarRocks has no TIME type; MySQL's ±838h range keeps its notation), `DATETIME` retains microsecond precision, `VARCHAR` character lengths are scaled ×4 for StarRocks' byte semantics (wide `CHAR` promotes to `VARCHAR`), `BIGINT UNSIGNED` maps to `LARGEINT` preserving the full unsigned range, `DECIMAL` precision is capped at 38, JSON/MAP map to `JSON`, and `BOOLEAN` maps to native `boolean`.
- **Metadata handling**: JDBC metadata lookups are qualified with the current database (the driver otherwise iterates every database and StarRocks errors on non-matching ones), and `BYTES` binding converts `ByteBuffer` values the driver cannot bind.
- **Catalog option** (dbz#979): `dialect.starrocks.catalog.name` executes `SET CATALOG <name>` on every new pooled connection, enabling writes into catalogs other than `default_catalog`; the driver URL's `<catalog>.<database>` form is documented as the primary path.
- **Tests & docs**: unit tests pin the dialect mappings and boundary cases; 108 integration tests (insert/delete/primary-key modes, CloudEvents, JSON, column type mappings including boundary values and writes into user-created `STRING` tables, OpenLineage) all pass against a StarRocks 4.1.1 Testcontainer; an e2e MySQL→StarRocks pipeline test and the `it-starrocks,e2e-starrocks` CI matrix entry are included, along with connector documentation covering the type mappings and limitations.

### Out of scope

- Geospatial and vector logical types, and mapping Kafka Connect `ARRAY`/`STRUCT` schemas to StarRocks `ARRAY`/`STRUCT` columns.
- StarRocks Stream Load based ingestion (this dialect uses the standard JDBC write path).
- `DECIMAL` precisions above StarRocks' maximum of 38.

### CODE REVIEW

File: debezium-connector-jdbc/src/main/java/io/debezium/connector/jdbc/JdbcSinkConnectorConfig.java [hunk 1]
```diff
@@ -585,6 +604,17 @@ public org.hibernate.cfg.Configuration getHibernateConfiguration() {
             hibernateConfig.setProperty(AvailableSettings.SHOW_SQL, Boolean.toString(true));
         }
 
+        if (!Strings.isNullOrBlank(starRocksCatalogName)) {
+            // The MySQL wire protocol has no notion of StarRocks catalogs, so the catalog is selected
+            // through an initial statement executed on every new pooled connection.
+            hibernateConfig.setProperty(AgroalSettings.AGROAL_CONFIG_PREFIX + ".initialSQL", "SET CATALOG " + starRocksCatalogName);
```
Comment (Naros) [inline debezium-connector-jdbc/src/main/java/io/debezium/connector/jdbc/JdbcSinkConnectorConfig.java:648]: Is there any issue with not quoting/escaping the catalog name here?

Comment (minleejae) [reply debezium-connector-jdbc/src/main/java/io/debezium/connector/jdbc/JdbcSinkConnectorConfig.java:648]: StarRocks executes multiple ;-separated statements per query, so an unescaped name here is an injection vector. Since valid catalog names are limited to [a-zA-Z_][a-zA-Z0-9_]* (no escaping ever needed), I added field validation that rejects anything else at config time. [3988e92](https://github.com/debezium/debezium/pull/7628/commits/3988e92f8a4764cd0f210a808499072695637211)

**Review by Naros — COMMENTED** [review]
Code LGTM, and its ready to merge, just waits on:

* https://github.com/debezium/debezium/pull/7411

**Comment by Naros** [comment]
Rebased on the latest changes from `main`.
